src/common/plugin/Artifacts.js

```diff
@@ -0,0 +1,35 @@
+/*globals define*/
+define([
+], function(
+) {
+    const Artifacts = function () {
+        throw new Error('Aritifacts is supposed to be used as an extension object, do not instantiate');
+    };
+
+    Artifacts.prototype.constructor = Artifacts;
+
+    Artifacts.prototype.getBaseNode = function (baseName) {
+        const metaDict = this.core.getAllMetaNodes(this.activeNode);
+        const metanodes = Object.keys(metaDict).map(id => metaDict[id]);
+        const base = metanodes.find(node =>
+            this.core.getAttribute(node, 'name') === baseName
+        );
+        return base;
+    };
+
+    Artifacts.prototype.createArtifact = async function (baseNode, attrs) {
```

Comment:
Why does this accept `baseNode` as an input? Isn't this always `Data` from the metamodel?